### DISCUSSION TITLE
docs/core/gdb-chaining-query-scan-and-count

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#74] [General]
# docs/core/gdb-chaining-query-scan-and-count

在使用GoFrame框架进行ORM查询时简化分页查询场景。通过ScanAndCount方法，可以在一次操作中完成数据查询和总数量查询，有效减少代码冗余，提高开发效率。适用于需要同时获取数据和其总数量的情况，如分页查询。

https://goframe.org/docs/core/gdb-chaining-query-scan-and-count

**Comment by wangle201210** [comment]
建议这里的用例里面不要加.OmitEmpty()
加在这里增加学习人员的理解成本
这里应该只需要说明怎么用ScanAndCount就行了

还有ScanAndCount这里第三个参数不是一般为false才合理么，分页一般是不看前面的Fields的

**Reply by gqcn** [reply]
done

**Comment by Mobs-demacia** [comment]
在以上示例中，如果获取的用户列表为空的时候，items会被赋值为nil，能否调整为返回空切片呢？

**Reply by Mobs-demacia** [reply]
items需要make一下，我疏忽了这一点，加上就可以了

**Comment by pengxinwen** [comment]
total参数可以从int改为int64吗

**Comment by jiguohui** [comment]
不错的方法，如果在集成一个SUM，就更好了。这样的话，列表、总数、总金额之类的，就都有了
